### PR TITLE
Add boundary for BigInteger.pow recursion

### PR DESCRIPTION
**Pull request by bjfish**
`BigInteger.pow` calls `BigInteger.square` recursively.

### CODE REVIEW

File: src/main/java/org/truffleruby/core/numeric/BignumNodes.java [hunk 1]
```diff
@@ -722,7 +722,7 @@ public DynamicObject pow(DynamicObject a, long b) {
                 return null; // Primitive failure
             } else {
                 // TODO CS 15-Feb-15 what about this cast?
-                return createBignum(Layouts.BIGNUM.getValue(a).pow((int) b));
+                return createBignum(pow(Layouts.BIGNUM.getValue(a),(int) b));
```
Comment (eregon) [inline src/main/java/org/truffleruby/core/numeric/BignumNodes.java]: Missing space after `,`

Comment (eregon) [inline src/main/java/org/truffleruby/core/numeric/BignumNodes.java]: Actually, all methods of BigInteger should be TruffleBoundary, except those that don't throw exceptions and have very few conditions. That is pretty much all methods except maybe `BigInteger.valueOf(long)`.

**Comment by eregon** [comment]
Actually, all methods of BigInteger should be TruffleBoundary, except those that don't throw exceptions and have very few conditions. That is pretty much all methods except maybe `BigInteger.valueOf(long)`.

File: src/main/java/org/truffleruby/core/numeric/BignumNodes.java (hunk 1 above)
Comment (bjfish) [inline src/main/java/org/truffleruby/core/numeric/BignumNodes.java]: I've added this space.

**Pull request merged**